global/scripts/debug.shlib

```diff
@@ -0,0 +1,45 @@
+# Description:
+#
+#   Function for printing debugging information on caught errors.
+#   It makes use of error trapping. To reset the trap, add
+#   end of the script using "trap - EXIT".
+#
+# Input:
+#
+#   — line number
+#   - command ran
+#   - exit status
+
+trap 'on_error $LINENO "$BASH_COMMAND" $?; exit' ERR
+
+on_error()
+{
+    me=`basename $0`
+    line=($2)
```

Comment:
I would at least put double quotes around `$0` since it is passed to basename.  It looks like `line=($2)` could be fooled by spaces in a folder name, but it isn't clear that bash provides a way around that.  I would probably not separate the command from its arguments if I can't guarantee it is done correctly.  Since the executable part of the command may itself contain a variable, perhaps it should just expand it as `echo "---> expanded: $(eval echo "$2")"` (yes, quotes do nest properly like that).

Reply:
I changed the expanded line to the suggested snippet.

Reply:
Done